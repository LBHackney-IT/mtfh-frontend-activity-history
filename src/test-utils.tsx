import { Router, Route } from 'react-router-dom';
import React from 'react';
import fetch from 'node-fetch';
import { rest } from 'msw';
import { createMemoryHistory, MemoryHistory } from 'history';
import { render, RenderResult } from '@testing-library/react';
import { queries } from '@mtfh/common';
import { server } from './mocks';

beforeAll(() => {
    server.listen();
});

afterEach(() => {
    server.resetHandlers();
});

afterAll(() => {
    server.close();
});

interface RouteRenderConfig {
    url: string;
    path: string;
    query: keyof typeof queries;
}

export const routeRender = (
    component: JSX.Element,
    options?: Partial<RouteRenderConfig>
): [RenderResult, MemoryHistory] => {
    const config: RouteRenderConfig = {
        url: '/person/be8c805c-b1de-11eb-8529-0242ac130003',
        path: '/person/:personId',
        query: 'lg',
        ...options,
    };
    const history = createMemoryHistory();
    history.push(config.url);
    return [
        render(
            <Router history={history}>
                <Route path={config.path}>{component}</Route>
            </Router>
        ),
        history,
    ];
};

export const get = (path: string, data: unknown, code = 200): void => {
    server.use(
        rest.get(path, (req, res, ctx) => {
            return res(ctx.status(code), ctx.json(data));
        })
    );
};
