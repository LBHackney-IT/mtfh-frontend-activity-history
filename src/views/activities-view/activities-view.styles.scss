.mtfh-activities-history {
    b {
        font-weight: 600;
    }

    p {
        margin: 0;
    }
}
