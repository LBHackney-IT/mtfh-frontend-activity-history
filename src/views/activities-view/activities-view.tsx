import { Link as RouterLink, useParams } from 'react-router-dom';
import React from 'react';
import { Button, Link } from '@mtfh/common';
import { locale, usePerson } from '@services';
import { ActivityHistoryListLegacy } from '@components';

const { pageTitle, closeButton } = locale.activities;

export interface EntityRequestId {
    id: string;
    entityType: string;
}

const PersonInformation = ({ id, entityType }: EntityRequestId) => {
    const { data: person } = usePerson(id);
    return (
        <>
            <Link
                as={RouterLink}
                to={`/${entityType}/${id}`}
                variant="back-link"
            >
                {person?.firstName} {person?.surname}
            </Link>
            <h1 className="lbh-heading-h1">{pageTitle}</h1>
            <h2 className="lbh-heading-h2">
                {person?.firstName} {person?.surname}
            </h2>
        </>
    );
};

export const ActivitiesViewLegacy = (): JSX.Element => {
    const { id, entityType } = useParams<{ id: string; entityType: string }>();

    return (
        <div data-testid="activities">
            <PersonInformation id={id} entityType={entityType} />
            <ActivityHistoryListLegacy targetId={id} />
            <Button
                as={RouterLink}
                to={`/${entityType}/${id}`}
                variant="secondary"
            >
                {closeButton}
            </Button>
        </div>
    );
};
