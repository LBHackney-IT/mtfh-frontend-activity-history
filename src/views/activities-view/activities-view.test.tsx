import React from 'react';
import { screen, waitFor } from '@testing-library/react';

import { locale } from '@services';

import { ActivitiesView } from '../activities-view';
import { routeRender } from '../../test-utils';

test('renders the activities view', async () => {
    const [{ container }] = routeRender(<ActivitiesView />);
    expect(container).toMatchSnapshot();

    await waitFor(() =>
        expect(screen.getAllByRole('heading')[0]).toHaveTextContent(
            locale.activities.pageTitle
        )
    );
});
