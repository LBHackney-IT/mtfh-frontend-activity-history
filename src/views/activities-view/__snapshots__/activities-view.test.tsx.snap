// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the activities view 1`] = `
<div>
  <div
    data-testid="activities"
  >
    <a
      class="govuk-back-link lbh-back-link lbh-link--no-visited-state"
      href="/undefined/undefined"
    >
       
    </a>
    <h1
      class="lbh-heading-h1"
    >
      Activity history
    </h1>
    <h2
      class="lbh-heading-h2"
    >
       
    </h2>
    <table
      class="govuk-table lbh-table"
    >
      <thead
        class="govuk-table__head"
      >
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__cell govuk-table__header"
          >
            date
          </th>
          <th
            class="govuk-table__cell govuk-table__header"
          >
            category
          </th>
          <th
            class="govuk-table__cell govuk-table__header"
          >
            edit details
          </th>
          <th
            class="govuk-table__cell govuk-table__header"
          >
            edited by
          </th>
        </tr>
      </thead>
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row govuk-table__row mtfh-activity-history"
        >
          <td
            class="govuk-table__cell"
          >
            <div>
              <p>
                19.09.19
              </p>
              <p>
                03:12
              </p>
            </div>
          </td>
          <td
            class="govuk-table__cell"
          >
            Edit to:
             
            person
          </td>
          <td
            class="govuk-table__cell"
          >
            <div>
              <p>
                <b>
                  Last name
                </b>
              </p>
              <p>
                Removed:
                 
                <b>
                  Baker
                </b>
              </p>
            </div>
          </td>
          <th
            class="govuk-table__cell govuk-table__header"
          >
            Paul Fox
          </th>
        </tr>
        <tr
          class="govuk-table__row govuk-table__row mtfh-activity-history"
        >
          <td
            class="govuk-table__cell"
          >
            <div>
              <p>
                19.09.19
              </p>
              <p>
                03:12
              </p>
            </div>
          </td>
          <td
            class="govuk-table__cell"
          >
            Edit to:
             
            person
          </td>
          <td
            class="govuk-table__cell"
          >
            <div>
              <p>
                <b>
                  First name
                </b>
              </p>
              <p>
                Added:
                 
                <b>
                  Susanna
                </b>
              </p>
            </div>
          </td>
          <th
            class="govuk-table__cell govuk-table__header"
          >
            Paul Fox
          </th>
        </tr>
        <tr
          class="govuk-table__row govuk-table__row mtfh-activity-history"
        >
          <td
            class="govuk-table__cell"
          >
            <div>
              <p>
                19.02.21
              </p>
              <p>
                03:12
              </p>
            </div>
          </td>
          <td
            class="govuk-table__cell"
          >
            Edit to:
             
            person
          </td>
          <td
            class="govuk-table__cell"
          >
            <div>
              <p>
                <b>
                  Title
                </b>
              </p>
              <p>
                Previously:
                 
                <b>
                  Ms
                </b>
              </p>
              <p>
                Changed to:
                 
                <b>
                  Mrs
                </b>
              </p>
            </div>
            <div>
              <p>
                <b>
                  Preferred Last name
                </b>
              </p>
              <p>
                Previously:
                 
                <b>
                  Baker
                </b>
              </p>
              <p>
                Changed to:
                 
                <b>
                  Georgia
                </b>
              </p>
            </div>
          </td>
          <th
            class="govuk-table__cell govuk-table__header"
          >
            Mary Smith
          </th>
        </tr>
      </tbody>
    </table>
    <a
      class="govuk-button lbh-button govuk-secondary lbh-button--secondary"
      href="/undefined/undefined"
    >
      Close activity history
    </a>
  </div>
</div>
`;
